packages/formdata-event/ts_src/environment_api/document.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * @license
+ * Copyright (c) 2020 The Polymer Project Authors. All rights reserved. This
+ * code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt The complete set of authors may be
+ * found at http://polymer.github.io/AUTHORS.txt The complete set of
+ * contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt Code
+ * distributed by Google as part of the polymer project is also subject to an
+ * additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import {methods as DocumentMethods} from '../environment/document.js';
+
+export const createElement = (
```

Comment:
You should be able to preserve the `localName` -> return type mapping with something like this:

```ts
export const createElement = <K extends keyof HTMLElementTagNameMap>(
  doc: Document,
  localName: K,
  options?: ElementCreationOptions,
): HTMLElementTagNameMap[K] => {
    return document.createElement.call(doc, localName, options) as HTMLElementTagNameMap[K];
};
```

And then you won't need the cast later on when you do `createElement('input')`.

Reply:
Very neat - added!